test/dynamo/test_modules.py

```diff
@@ -2221,18 +2221,21 @@ def foo(mod, x):
 
         mod = Mod()
         foo(mod, torch.rand([4]))
-        self.assertEqual(compiles_without_buffers, 0)
+        print(mod.buffer.storage().nbytes())
+        self.assertEqual(compiles_without_buffers, 1)
+        print(mod.buffer.dtype)
+        print(mod.buffer.storage().nbytes())
 
         foo(mod, torch.rand([4], dtype=torch.half))
-        self.assertEqual(compiles_without_buffers, 1)
+        self.assertEqual(compiles_without_buffers, 2)
 
         class Mod2(Mod):
             def __setattr__(self, name, value):
                 return super().__setattr__(name, value)
 
         foo(Mod2(), torch.rand([4]))
         # causes two compilations, bc unimplemented custom setattr
-        self.assertTrue(compiles_without_buffers >= 2)
+        self.assertTrue(compiles_without_buffers >= 4)
```

Comment:
@eellison could you confirm that this is the actual intended behaviour of your test as introduced in https://github.com/pytorch/pytorch/pull/106092?